tests/API/Unit/Baggage/Propagation/BaggagePropagatorTest.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Tests\API\Unit\Baggage\Propagation;
+
+use OpenTelemetry\API\Baggage\Baggage;
+use OpenTelemetry\API\Baggage\Metadata;
+use OpenTelemetry\API\Baggage\Propagation\BaggagePropagator;
+use OpenTelemetry\Context\Context;
+use PHPUnit\Framework\TestCase;
+
+class BaggagePropagatorTest extends TestCase
+{
+    public function test_fields(): void
+    {
+        $this->assertSame(
+            ['baggage'],
+            BaggagePropagator::getInstance()->fields()
+        );
+    }
+
+    public function test_inject_emptyBaggage(): void
+    {
+        $carrier = [];
+
+        BaggagePropagator::getInstance()->inject($carrier);
+
+        $this->assertEmpty($carrier);
+    }
+
+    public function test_inject(): void
+    {
+        $carrier = [];
+
+        BaggagePropagator::getInstance()->inject(
+            $carrier,
+            null,
+            Context::getRoot()->withContextValue(
+                Baggage::getBuilder()
+                    ->set('nometa', 'nometa-value')
+                    ->set('meta', 'meta-value', new Metadata('somemetadata; someother=foo'))
+                    ->build()
+            )
+        );
+
+        $this->assertSame(
+            ['baggage' => 'nometa=nometa-value,meta=meta-value;somemetadata; someother=foo'],
+            $carrier
+        );
+    }
+
+    public function test_extract_missingHeader(): void
+    {
+        $this->assertEquals(
+            Context::getRoot(),
+            BaggagePropagator::getInstance()->extract([])
+        );
+    }
+
+    /** @dataProvider headerProvider */
+    public function test_extract(string $header, Baggage $expectedBaggage): void
+    {
+        $propagator = BaggagePropagator::getInstance();
+
+        $context = $propagator->extract(['baggage' => $header]);
+
+        $this->assertEquals(
+            $expectedBaggage,
+            Baggage::fromContext($context)
+        );
+    }
+
+    public function headerProvider(): array
```

Comment:
Nice helper function! TIL about `@dataProvider` in phpunit; that would have been super convenient to know about a bunch of different times 😺